### PR TITLE
Move towards MySQL 8.0 as the default template generation (#11153)

### PR DESCRIPTION
**Pull request by rsajwani**
* Move towards MySQL 8.0 as the default template generation

This upgrades the remaining things to Ubuntu 20.04 and makes MySQL 8.0 the default we run tests against. We still have tests for MySQL 5.7 but those are now explicitly opted into.

This should finish up the Ubuntu 20.04 upgrade and also makes things easier for the future when we need to upgrade again.

Signed-off-by: Dirkjan Bussink <d.bussink@gmail.com>

* CI: rename shard vtorc_8.0 to vtorc_5.7, change expected test output for 8.0

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: increase timeout for 8.0 tests on the actual test step from 30 to 45 mins

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: increase timeout to 45 minutes for mysql57 tests too. We really only need this for vtorc, but I've made the change to the template so all tests get it.

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: fix vtorc test to work with both 5.7 and 8.0

Signed-off-by: deepthi <deepthi@planetscale.com>

* CI: missed docker flag in mysql57 template, one more fix to vtorc test

Signed-off-by: deepthi <deepthi@planetscale.com>

* removing spaces from pb file

Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

* removing spaces in pb file part 2

Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

Signed-off-by: Dirkjan Bussink <d.bussink@gmail.com>
Signed-off-by: deepthi <deepthi@planetscale.com>
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>
Co-authored-by: deepthi <deepthi@planetscale.com>
Co-authored-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description

There are two main changes in CI workflow been done as part of this PR.
- Move to ubuntu 20.04
- Upgrade Mysql 5.7 to Mysql 8.0.25

This PR cherry-pick (#11153) the change from main where we have made Mysql8.0.x as default database (from Mysql 5.7). There were certain issues which were hard to cherry-pick from main as a result of moving Mysql8.0, especially failures in onlineddl_vrepl_suite which compelled us to pin Mysql version to 8.0.25. For example we no longer use Tengo whereas in v13 we used it for schemadiff. To remove Tango from v13 is a bigger work item.

Installing Deb files for 8.0.25 was giving a lot of pain during requisite deployment, therefore, we choose to un-tar the package with binaries and set VT_MYSQL_ROOT directory during `run end-to-end step` in workflow. 

Apart from workflow changes, there were some compatibility issues between Mysql5.7 and Mysql8.0 which resulted in test failures. They are also addressed in this PR.

## Related Issue(s)
https://github.com/vitessio/vitess/issues/11511

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletserver/tabletserver.go [hunk 1]
```diff
@@ -1399,9 +1399,10 @@ func (tsv *TabletServer) convertAndLogError(ctx context.Context, sql string, bin
...
 
 // truncateSQLAndBindVars calls TruncateForLog which:
-//  splits off trailing comments, truncates the query, re-adds the trailing comments,
-//  if sanitize is false appends quoted bindvar:value pairs in sorted order, and
-//  lastly it truncates the resulting string
+//
+//	splits off trailing comments, truncates the query, re-adds the trailing comments,
+//	if sanitize is false appends quoted bindvar:value pairs in sorted order, and
+//	lastly it truncates the resulting string
 func truncateSQLAndBindVars(sql string, bindVariables map[string]*querypb.BindVariable, sanitize bool) string {
```
Comment (harshit-gangal) [inline go/vt/vttablet/tabletserver/tabletserver.go:1406]: not sure about this formatting change, what would have caused this.

**Review by harshit-gangal — APPROVED** [review]
LGTM for query serving side changes.

**Comment by rohit-nayak-ps** [comment]
@mattlord, I noticed we have this commit (related to the messaging deadlock) in this issue: https://github.com/vitessio/vitess/pull/11619/commits/6f0b5697c95d24e5807af5ac1bd491f91af455cf. Is that affected by switching to mysql 8.0 for testing?

**Review by rohit-nayak-ps — APPROVED** [review]
vrep related changes look good.

**Pull request merged**